libnavigation-core/src/test/java/com/mapbox/navigation/core/accounts/MapboxNavigationAccountsTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.mapbox.navigation.core.accounts
+
+import android.content.Context
+import android.os.Bundle
+import androidx.test.core.app.ApplicationProvider
+import com.mapbox.android.accounts.v1.AccountsConstants
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class MapboxNavigationAccountsTest {
+
+    val ctx = ApplicationProvider.getApplicationContext<Context>()
+
+    @Before
+    fun setUp() {
+        ctx.applicationInfo.metaData = Bundle().also {
+            it.putBoolean(AccountsConstants.KEY_META_DATA_MANAGE_SKU, false)
+        }
+
+        ctx.getSharedPreferences(
+                AccountsConstants.MAPBOX_SHARED_PREFERENCES,
+                Context.MODE_PRIVATE
+        ).edit().putString("com.mapbox.navigation.accounts.trips.skutoken", "myTestToken")
+                .commit()
+
+        ctx.getSharedPreferences(
+                AccountsConstants.MAPBOX_SHARED_PREFERENCES,
+                Context.MODE_PRIVATE
+        ).edit().putString("com.mapbox.navigation.accounts.mau.skutoken", "myTestToken")
+                .commit()
+    }
+
+    @After
+    fun tearDown() {
+        ctx.applicationInfo.metaData = null
+    }
+
+    @Test(expected = IllegalStateException::class)
+    fun obtainSkuToken_when_resourceUrl_empty() {
+        val instance = MapboxNavigationAccounts.getInstance(ctx)
+
+        val result = instance.obtainUrlWithSkuToken("", 4)
+
+        assertEquals("", result)
```

Comment:
If we're expecting an exception this is not necessary. Actually, we're not adding the `assert` in `obtainSkuToken_when_resourceUrl_notNullOrEmpty_and_querySize_lessThan_zero` 😅

Reply:
As a side note, I'm not personally a fan of testing exceptions this way because `A`ssert part from triple A is gone, but that's personal preference and we can discuss further as a team.